Source/Plugins/EditorBase/PropertyEditors/AddComponentPropertyEditor.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using AdamsLair.WinForms.PropertyEditing;
+using Duality.Editor.Forms;
+using Duality.Editor.UndoRedoActions;
+using ButtonState = AdamsLair.WinForms.Drawing.ButtonState;
+
+namespace Duality.Editor.Plugins.Base.PropertyEditors
+{
+	/// <summary>
+	/// A <see cref="PropertyEditor"/> that provides a button
+	/// for adding a component to a <see cref="GameObject"/>
+	/// </summary>
+	public class AddComponentPropertyEditor : PropertyEditor
+	{
+		private Rectangle	rectButton		= Rectangle.Empty;
+		private bool		buttonHovered	= false;
+		private bool		buttonPressed	= false;
+
+		public override object DisplayedValue
+		{
+			get { return this.GetValue(); }
+		}
+
+		public AddComponentPropertyEditor()
+		{
+			this.Hints = HintFlags.None;
+			this.Height += 4;
+		}
+
+		protected override void UpdateGeometry()
+		{
+			base.UpdateGeometry();
+
+			this.rectButton = new Rectangle(
+				this.ClientRectangle.X + 2, this.ClientRectangle.Y + 2,
+				this.ClientRectangle.Width - 4, this.ClientRectangle.Height - 4);
```

Comment:
This is more of a style thing, but can you check how it looks when increasing padding on top and bottom of the editor? Maybe something between 5 and 20 pixels both top and bottom. Ignore if it doesn't turn out to look better.